File: backend/app/schemas/event.py
from pydantic import BaseModel, EmailStr, ConfigDict, Field
from app.models.event import StateEnum
from datetime import datetime
from app.schemas.session import SessionResponse


class EventCreate(BaseModel):
    name: str
    description: str
    capacity: int
    state: StateEnum
    date_start: datetime
    date_end: datetime
    location: str
    user_created_id: int
    session_ids: list[int] = Field(min_length=1)


class EventUpdate(BaseModel):
    name: str
    description: str
    capacity: int
    state: StateEnum
    date_start: datetime
    date_end: datetime
    location: str
    session_ids: list[int] = Field(min_length=1)


class EventResponse(BaseModel):
    id: int
    name: str
    description: str
    capacity: int
    state: StateEnum
    date_start: datetime
    date_end: datetime
    location: str
    user_created_id: int
    sessions: list[SessionResponse] = []  # <-- Aquí incluyes las sesiones asociadas
    is_registered: bool = False

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/app/event-managment/event.ts
import { Session } from '../session-managment/session';

export class Event {
  id: number;
  name: string;
  description: string;
  capacity: number;
  state: string;
  date_start: string;
  date_end: string;
  location: string;
  user_created_id: number;
  sessions: Session[];
  is_registered: boolean;

  constructor(
    id: number,
    name: string,
    description: string,
    capacity: number,
    state: string,
    date_start: string,
    date_end: string,
    location: string,
    user_created_id: number,
    sessions: Session[],
    is_registered: boolean
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.capacity = capacity;
    this.state = state;
    this.date_start = date_start;
    this.date_end = date_end;
    this.location = location;
    this.user_created_id = user_created_id;
    this.sessions = sessions;
    this.is_registered = is_registered
  }
}
